### DISCUSSION TITLE
User-generated content?

### DISCUSSION THREAD

**Discussion by ColeDCrawford** [tinacms/tinacms#3801] [Help]
Are there any good patterns for user-generated content? I'd like frontend users to be able to submit a form with a couple questions and potentially an image, and have that end up in Tina Cloud as a content type with a draft flag for review and possible publishing.

**Comment by jeffsee55** [comment]
One problem with this is that Tina Cloud doesn't support long-lived writable tokens (our tokens are read-only, so mutations need to go through auth). You could instead try it with [self-hosting](https://tina.io/docs/self-hosted/overview/), and you'd be responsible for authenticating your users with your own auth solution, there may be licensing issues with this though

Alternatively, if you're providing your own form to your users, as long as those submissions made their way into your Github repo Tina Cloud would pick them up as any other content. So you could write to a "submission" collection (eg. `content/submissions/user-submission.md`) with [Octokit](https://github.com/octokit/core.js/), you'd just need to ensure it writes the markdown or json in the shape that your Tina collection expects.

**Comment by scottgallant** [comment]
@ColeDCrawford, see the self-hosted license [here](https://github.com/tinacms/tinacms/blob/main/packages/%40tinacms/datalayer/LICENSE) and related [FAQ](https://github.com/tinacms/tinacms/blob/main/packages/%40tinacms/datalayer/LICENSE-FAQ.md#what-is-a-user-in-the-tina-data-layer-license).  We'd be happy to do a custom license if that doesn't fit for you. Feel free to contact me (firstname at tina.io).
